invenio_rdm_records/fixtures/vocabularies.py

```diff
@@ -106,84 +109,249 @@ def __iter__(self):
 
 
 #
-# Fixture
+# Exceptions
 #
-class VocabulariesFixture:
-    """Vocabularies fixture."""
+class ConflictingFixturesError(Exception):
+    """Exception when multiple modules provide same subject."""
 
-    def __init__(self, identity, search_paths, filename):
-        """Initialize the fixture."""
-        self._search_paths = search_paths
-        self._filename = filename
+    def __init__(self, errors):
+        """Constructor."""
+        message = "\n".join(errors)
+        super().__init__(message)
+        self.errors = errors
+
+
+#
+# Fixture
+#
+class VocabulariesFixtures:
+    """Concept of Vocabulary fixtures across locations.
+
+    This class' responsibility is to load different vocabularies in
+    this priority order: app_data, then extensions and then this package.
+
+    Earlier vocabularies in this hierarchy are chosen first. But in the case
+    of subvocabularies (e.g. subject types), having loaded a subvocabulary
+    before, shouldn't prevent loading another one. Only if the same
+    subvocabulary is encountered again should it be ignored.
+
+    Concretely, having loaded MeSH subjects shouldn't prevent loading FAST
```

Comment:
sanity check: imagine an extension is loading some subjects (e.g. mesh) and the user doesn't want it. the way to exclude it is by not installing `invenio-mesh`, and since there is no default (at package level) the user can fully customize his/her choice by modifying the `vocabularies.yaml` at instance (cookiecutter) level and/or adding new extensions.

Reply:
> the way to exclude it is by [...]

Yes there are a couple way for the administrator to be in full control in this situation. They can either:

1- not install invenio-subjects-mesh in the first place (there is no default MeSH subjects so they won't get any)
2- define in `app_data/vocabularies.yaml` a `subjects` `mesh` datafile that points to an empty yaml file. The loading priority will make sure that the later defined one is ignored. Obviously 1- is preferable. 

They can install other extensions to get other subjects or again just stick a specific subjects yaml file in their `app_data/`(e.g. if they don't want the overhead of creating an extension - not something shareable with other instances).